src/reader.cpp

```diff
@@ -1206,6 +1232,58 @@ void reader_data_t::maybe_refilter_pager(const editable_line_t *el) {
     }
 }
 
```

Comment:
trying to dogfood now with

	function fish_user_key_bindings
	    bind -e --preset \cn
	    bind -e --preset \cp
	end
	set -g fish_autosuggestion_enabled 0

but it's hard getting used to, some details seem off.
I wonder if it should be closer to other shells (execute command immediately on enter, movement keys cancel search) but that seems backwards and would prevent us from using pager features